strawberryfields/ops.py

```diff
@@ -2796,11 +2906,17 @@ def _decompose(self, reg, **kwargs):
     Catstate,
     Thermal,
 )  # have __init__ methods with default arguments
-state_preparations = simple_state_preparations + (Ket, DensityMatrix)
+state_preparations = simple_state_preparations + (Ket, DensityMatrix, Bosonic, GKP)
```

Comment:
Just curious for my own understanding: `Gaussian` can be decomposed, could `Bosonic` be also decomposed? (Or that would be more complex.)

Reply:
When you say `Gaussian` can be decomposed, do you mean that you can write a general gaussian preparation in terms of interferometers and squeezers? If that's what you mean, then no, Bosonic cannot be decomposed in the same way. I am not aware of a general set of operations that can that can decompose a non-Gaussian state preparation.

Reply:
Thanks! Yes, was considering how [the `Gaussian` operation is decomposed in SF](https://strawberryfields.readthedocs.io/en/stable/code/api/strawberryfields.ops.Gaussian.html?highlight=gaussian#strawberryfields.ops.Gaussian).